# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/DrawTargetWebgl.cpp
# Commit: c5112a80ad2f
# Full Hash: c5112a80ad2f910f06889c9cadecc8c4728cd858
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-11-08 21:36:02
# Description:
#   Bug 1799495 - Ensure WaitForShmem has a valid DrawTargetWebgl to access. r=aosmond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161554
# ==============================================================================

diff -r 879b0989791c -r c5112a80ad2f dom/canvas/DrawTargetWebgl.cpp
--- a/dom/canvas/DrawTargetWebgl.cpp	Tue Nov 08 16:21:48 2022 +0000
+++ b/dom/canvas/DrawTargetWebgl.cpp	Tue Nov 08 16:23:56 2022 +0000
@@ -224,7 +224,7 @@
       // Force any Skia snapshots to copy the shmem before it deallocs.
       mSkia->DetachAllSnapshots();
       // Ensure we're done using the shmem before dealloc.
-      mSharedContext->WaitForShmem();
+      mSharedContext->WaitForShmem(this);
       auto* child = mSharedContext->mWebgl->GetChild();
       if (child && child->CanSend()) {
         child->DeallocShmem(mShmem);
@@ -3226,7 +3226,7 @@
   mSkia->FillGlyphs(aFont, aBuffer, aPattern, aOptions);
 }
 
-void DrawTargetWebgl::SharedContext::WaitForShmem() {
+void DrawTargetWebgl::SharedContext::WaitForShmem(DrawTargetWebgl* aTarget) {
   if (mWaitForShmem) {
     // GetError is a sync IPDL call that forces all dispatched commands to be
     // flushed. Once it returns, we are certain that any commands processing
@@ -3235,7 +3235,9 @@
     mWaitForShmem = false;
     // The sync IPDL call can cause expensive round-trips to add up over time,
     // so account for that here.
-    mCurrentTarget->mProfile.OnReadback();
+    if (aTarget) {
+      aTarget->mProfile.OnReadback();
+    }
   }
 }
 